packages/aws-cdk-lib/core/lib/cfn-fn.ts

```diff
@@ -500,9 +502,31 @@ class FnFindInMap extends FnBase {
    * @param mapName The logical name of a mapping declared in the Mappings section that contains the keys and values.
    * @param topLevelKey The top-level key name. Its value is a list of key-value pairs.
    * @param secondLevelKey The second-level key name, which is set to one of the keys from the list assigned to TopLevelKey.
+   * @param defaultValue The value of the default value returned if either the key is not found in the map
    */
-  constructor(mapName: string, topLevelKey: any, secondLevelKey: any) {
-    super('Fn::FindInMap', [mapName, topLevelKey, secondLevelKey]);
+
+  private readonly mapName: string;
+  private readonly topLevelKey: string;
+  private readonly secondLevelKey: string;
+  private readonly defaultValue?: string;
+
+  constructor(mapName: string, topLevelKey: any, secondLevelKey: any, defaultValue?: string) {
+    super('Fn::FindInMap', [mapName, topLevelKey, secondLevelKey, defaultValue !== undefined ? { DefaultValue: defaultValue } : undefined]);
+    this.mapName = mapName;
+    this.topLevelKey = topLevelKey;
+    this.secondLevelKey = secondLevelKey;
+    this.defaultValue = defaultValue;
+  }
+
+  public resolve(context: IResolveContext): any {
+    if (this.defaultValue !== undefined) {
+      Stack.of(context.scope).addTransform('AWS::LanguageExtensions');
+    }
```

Comment:
This could also be located in the `resolve()` function of the `CfnMappingEmbedder` class. It would remove the need for assigning the 4 variables in this class. However, if it is there, using `Fn.findInMap` would not work with the default value. Only `CfnMapping.findInMap` could use the functionality.

Reply:
👍 this is the better location